### PR TITLE
vfs: fix incorrect st_ino in entry.attr

### PR DESCRIPTION
**Pull request by sctb512**
At present, the entry.inode is the correct inode number, but the linux kernel uses the entry.attr.st_ino as inode number. So, we should correct the value of entry.attr.st_ino to let it work.

This PR is related to https://github.com/dragonflyoss/image-service/issues/851

Result:
c4106f14-cfdf-44cc-8082-3822f38b4186

Signed-off-by: Bin Tang <tangbin.bin@bytedance.com>

### CODE REVIEW

File: src/api/vfs/sync_io.rs [hunk 1]
```diff
@@ -119,7 +126,14 @@ impl FileSystem for Vfs {
     ) -> Result<(stat64, Duration)> {
         match self.get_real_rootfs(inode)? {
             (Left(fs), idata) => fs.setattr(ctx, idata.ino(), attr, handle, valid),
-            (Right(fs), idata) => fs.setattr(ctx, idata.ino(), attr, handle, valid),
+            (Right(fs), idata) => {
+                let ino = self.convert_inode(idata.fs_idx(), idata.ino())?;
+                fs.setattr(ctx, idata.ino(), attr, handle, valid)
+                    .map(|(mut attr, duration)| {
+                        attr.st_ino = ino;
```
Comment (bergwolf) [inline src/api/vfs/sync_io.rs:133]: `idata` is already an `VfsInode`. Something like
```
attr.st_ino = idata.into();
```
should suffice.

File: src/api/vfs/sync_io.rs [hunk 2]
```diff
@@ -105,7 +105,14 @@ impl FileSystem for Vfs {
     ) -> Result<(stat64, Duration)> {
         match self.get_real_rootfs(inode)? {
             (Left(fs), idata) => fs.getattr(ctx, idata.ino(), handle),
-            (Right(fs), idata) => fs.getattr(ctx, idata.ino(), handle),
+            (Right(fs), idata) => {
+                let ino = self.convert_inode(idata.fs_idx(), idata.ino())?;
+                fs.getattr(ctx, idata.ino(), handle)
+                    .map(|(mut attr, duration)| {
+                        attr.st_ino = ino;
```
Comment (bergwolf) [inline src/api/vfs/sync_io.rs:112]: Same here.

**Review by bergwolf — COMMENTED** [review]
Otherwise lgtm!

File: src/api/vfs/sync_io.rs (hunk 1 above)
Comment (sctb512) [reply src/api/vfs/sync_io.rs]: Fixed, it works. PTAL

**Review by bergwolf — APPROVED** [review]
Thanks!

**Pull request merged**